src/operators/Remap/avtRemapFilter.C

```diff
@@ -221,7 +230,24 @@ avtRemapFilter::Execute(void)
     // ------------------------------------ //
 
     avtDataTree_p inTree = GetInputDataTree();
-    TraverseDomainTree(inTree);
+    int totalNodes;
+    vtkDataSet **dataSets = inTree->GetAllLeaves(totalNodes);
+    for (int i = 0; i < totalNodes; ++i)
+    {
+        ClipDomain(dataSets[i]);
+    }
+
+#ifdef PARALLEL
+    int size = vars->GetNumberOfTuples();
+    double *vars_double = (double*) vars->GetVoidPointer(0);
+    double *new_buff = new double[size];
+    SumDoubleArrayAcrossAllProcessors(vars_double, new_buff, size);
+    for (int i = 0; i < size; ++i)
+    {
+        vars_double[i] = new_buff[i];
+    }
+    delete [] new_buff;
+#endif
```

Comment:
You should only have processor 0 return a non empty avtDataTree. So you only need to copy the values over for processor 0. You can delete the rectilinear grid and return an empty avtDataTree on the other ranks. The logic would be something like

if (PAR_Rank() == 0)
{
    SetOutputDataTree(new avtDataTree(rg, 0);
}
else
{
    rg->Delete();
    SetOutputDataTree(new avtDataTree(NULL, -1);
}

Reply:
I've attempted to do this and ran into some issues. I experience the following:
1. Load dataset `ghost1` and attempt to remap `zvar`
1. Error message: no input was defined for this filter.
1. Click draw again. Because the processor deleted `rg`, the error is now `engine exited abnormally`. The processor crashes
1. Click draw again. The remap will draw, but it is only serial this time.

I also tried it without the `rg->Delete()`. Still got the error message for no input was defined for this filter, no matter how many times I clicked "draw".

Any thoughts?

Reply:
> You should only have processor 0 return a non empty avtDataTree.

@brugger1 Can you explain why only rank 0 should have the data?

Reply:
You are remapping the data onto a single block, so there should only be a single vtk data set. That is the one on rank 0. If each rank had a copy of the data, you would have N redundant copies of the same data set. This will lead to rendering the data N times.

Reply:
I experimented with the code I gave you and it wasn't quite right. The following works.

    if (PAR_Rank() == 0)
    {
        avtDataTree_p newtree = new avtDataTree(rg, 0);
        SetOutputDataTree(newtree);
    }
    else
    {
        avtDataTree_p newtree = new avtDataTree();
        SetOutputDataTree(newtree);
    }